### ISSUE TITLE
Cannot open a file when filepath contains space on Mac

### ISSUE THREAD

**Issue by iceout** [neovide/neovide#1399]
**Describe the bug**
On Mac, execute such command `neovide "/FILE/PATH WITH SPACE/filename"`, only open an empty file.

**To Reproduce**
1. Open a terminal
2. execute `neovide "/FILE/PATH WITH SPACE/filename"`
3. See neovide open an empty buffer

**Expected behavior**
Neovide should open  the selected file

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS 12.4
 - Neovide Version: 0.8.0
 - Neovim Version: 0.7.0

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
The log file is too long to post here.

**Additional context**
nvim can open such "/FILE/PATH WITH SPACE/filename" file

**Comment by aJuvan** [comment]
I am experiencing the same issue.

Looking through the logs and source I noticed that the arguments get passed to `nvim` [by inserting them into a shell command](https://github.com/neovide/neovide/blob/48dfe95b6e3b2552afbf805921aeeb26cf124d05/src/bridge/command.rs#L117), which causes them to get parsed again. This causes the `neovide 'path with spaces/file.txt'` to be passed as `["/bin/zsh", "-lc", "nvim --embed path with spaces/file.txt"]` and furthermore, if you for example have files/folder with special characters, those also get "re-parsed" incorrectly (i.e. `neovide 'pathwith$symbol/file.txt`).

One fix would be to just omit the whole check for `macos` and let the [below `else` scope](https://github.com/neovide/neovide/blob/48dfe95b6e3b2552afbf805921aeeb26cf124d05/src/bridge/command.rs#L119) handle the `nvim` binary and it's arguments correctly (successfully tested on an M1 macbook).

If that can't be changed for some reason, the arguments should first be escaped (i.e. [shell-escape](https://docs.rs/shell-escape/latest/shell_escape/)).

A quick and dirty fix that also worked while testing was to just wrap the files with `'` by changing the [linked line](https://github.com/neovide/neovide/blob/48dfe95b6e3b2552afbf805921aeeb26cf124d05/src/cmd_line.rs#L173) to:
```sh
neovim_args.extend::<Vec<String>>(files_to_open.iter().map(|file| format!("'{}'", file).to_string()).collect());
```
But this change will most definitely lead to unintended consequences in the above mentioned `else` clause, so it should not be used.

**Comment by MultisampledNight** [comment]
Fixed through #1415, thank you both!
